src/core/or/command.c

```diff
@@ -535,6 +542,7 @@ command_process_relay_cell(cell_t *cell, channel_t *chan)
                (unsigned)cell->circ_id);
       if (CIRCUIT_IS_ORIGIN(circ)) {
         circuit_log_path(LOG_WARN, LD_OR, TO_ORIGIN_CIRCUIT(circ));
+        control_event_circ_bandwidth_used_for_circ(TO_ORIGIN_CIRCUIT(circ));
```

Comment:
We seem to be calling this function from three places. How can we know that these are the only places we need to call it from?

Reply:
We call it when we get an error to close the circuit, or when the codepaths called from circuit_receive_relay_cell() fails to update bw values (because the cell was not valid).

Comment 02b5a6c67f2d3287ffd2d5ca773ff3534d1c774c also clarifies this.